modules/users/client/components/Avatar.component.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import has from 'lodash/has';
+import '@/config/lib/i18n';
+import { withNamespaces } from 'react-i18next';
+
+/**
+ * User's avatar
+ * @param {object} user - User object
+ * @param {integer=256} size - Size of the image <img>. Supported values are 2048, 1024, 512, 256, 128, 64, 36, 32, 24 and 16. See avatar.less for details. Defaults to 256.
+ * @param {string=''} source - Leave empty to use user's selected source. Values "none", "facebook", "local", "gravatar".
+ * @param {boolean=true} link - Include a link to user's profile. Defaults to true.
+ */
+export function Avatar({ t, user, size=256, source='', link=true }) {
+
+  source = source || user.avatarSource;
+  const defaultAvatar = '/img/avatar.png';
+
+  const avatar = avatarUrl(user, source, size, defaultAvatar);
+
+  const img = (
+    <img className={`avatar avatar-${size} avatar-${source}`}
+      alt={''/* t('Profile picture of {{name}}', { name: user.displayName }) */}
+      aria-hidden="true"
+      src={avatar}
+      draggable="false"
+    />
+  );
+
+  return (
+    <div>{
+      (link)
+        ?
+        <a
+          href={`/profile/${user.username}`}
+          aria-label={t('Open user profile for {{name}}', { name: user.displayName })}
+        >{img}</a>
+        :
+        img
+    }</div>
+  );
+}
+
+const AvatarHOC = withNamespaces('user')(Avatar);
+
+AvatarHOC.propTypes = {
+  user: PropTypes.object.isRequired,
+  size: PropTypes.number,
+  source: PropTypes.string,
+  link: PropTypes.bool
+};
+
+Avatar.propTypes = {
+  ...AvatarHOC.propTypes,
+  t: PropTypes.func.isRequired
+};
+
+Object.defineProperty(AvatarHOC, 'name', { value: 'Avatar' });
+
+export default AvatarHOC;
```

Comment:
To me, this structure with component+translation HOC reads a bit funky and is confusing because Avatar is now named as the HOC, not the translation HOC.

I'm guessing `defineProperty` trick is to give it a name for React devtools to recognise it?

If we start copying this pattern elsewhere, it'll have an impact on code readability and potentially make enchancing components by composing multiple HOCs more complicated:
```js
export default compose(
	withNamespaces,
	someOtherHoc
)( Avatar );
```

Would something from [these examples work?](https://react.i18next.com/components/withnamespaces#sample)

My favourite, simply:
```js
export default withNamespaces('user')(Avatar);
```

I think it's okay if `t` isn't covered by propTypes?